src/Http/Http.Extensions/test/RequestDelegateGenerator/CompileTimeGeneratorTests.cs

```diff
@@ -0,0 +1,84 @@
+
+        // Emits diagnostic but generates no source
+        var result = Assert.IsType<GeneratorRunResult>(generatorRunResult);
+        var diagnostic = Assert.Single(result.Diagnostics);
+        Assert.Equal(DiagnosticDescriptors.GetUnableToResolveParameterDescriptor("provider").Id, diagnostic.Id);
+        Assert.Empty(result.GeneratedSources);
+
+        // Falls back to runtime-generated endpoint
+        var endpoint = GetEndpointFromCompilation(compilation, false);
+
+        var httpContext = CreateHttpContext();
+        await endpoint.RequestDelegate(httpContext);
+        await VerifyResponseBodyAsync(httpContext, expectedBody);
+    }
+
+    [Fact]
+    public async Task MapGet_WithRequestDelegate_DoesNotGenerateSources()
+    {
+        var (generatorRunResult, compilation) = await RunGeneratorAsync("""
+app.MapGet("/hello", (HttpContext context) => Task.CompletedTask);
+""");
+        var results = Assert.IsType<GeneratorRunResult>(generatorRunResult);
+        Assert.Empty(GetStaticEndpoints(results, GeneratorSteps.EndpointModelStep));
+
+        var endpoint = GetEndpointFromCompilation(compilation, false);
+
+        var httpContext = CreateHttpContext();
+        await endpoint.RequestDelegate(httpContext);
+        await VerifyResponseBodyAsync(httpContext, "");
+    }
+
+    // Todo: Move this to a shared test that checks metadata once that is supported
+    // in the source generator.
+    [Theory]
+    [InlineData(@"app.MapGet(""/"", () => Console.WriteLine(""Returns void""));", null)]
+    [InlineData(@"app.MapGet(""/"", () => TypedResults.Ok(""Alright!""));", null)]
+    [InlineData(@"app.MapGet(""/"", () => Results.NotFound(""Oops!""));", null)]
+    [InlineData(@"app.MapGet(""/"", () => Task.FromResult(new Todo() { Name = ""Test Item""}));", "application/json")]
+    [InlineData(@"app.MapGet(""/"", () => ""Hello world!"");", "text/plain")]
+    public async Task MapAction_ProducesCorrectContentType(string source, string expectedContentType)
+    {
+        var (result, compilation) = await RunGeneratorAsync(source);
+
+        VerifyStaticEndpointModel(result, endpointModel =>
+        {
+            Assert.Equal("/", endpointModel.RoutePattern);
+            Assert.Equal("MapGet", endpointModel.HttpMethod);
+            Assert.Equal(expectedContentType, endpointModel.Response.ContentType);
+        });
+    }
+
+    [Fact]
+    public async Task MapAction_ExplicitRouteParamWithInvalidName_SimpleReturn()
+    {
+        var source = $$"""app.MapGet("/{routeValue}", ([FromRoute(Name = "invalidName" )] string parameterName) => parameterName);""";
+        var (_, compilation) = await RunGeneratorAsync(source);
+        var endpoint = GetEndpointFromCompilation(compilation);
+
+        var httpContext = CreateHttpContext();
+
+        var exception = await Assert.ThrowsAsync<InvalidOperationException>(() => endpoint.RequestDelegate(httpContext));
```

Comment:
RDF would just try to read `context.Request.RouteValues["invalidName"]` even though it's not part of the pattern, right? I know this PR is just about tests and not fixing behavior, but do we agree that we should align behavior here?

Reply:
RDF throws the exception earlier when the endpoint is being constructed ([ref](https://github.com/dotnet/aspnetcore/blob/ffa0ec303abea99226d991c178c571a439fb4cbf/src/Http/Http.Extensions/src/RequestDelegateFactory.cs#L676-L679)). We could do something similar in RDG, but that would mean presenting the exception at compile-time which would require us to have a way to parse the route patterns in the generator. https://github.com/dotnet/aspnetcore/issues/46342 tracks doing this work.